src/ica_api.c

```diff
@@ -1350,17 +1350,17 @@ int ica_ec_key_init(const unsigned char *X, const unsigned char *Y,
 {
 	unsigned int privlen;
 
+	/* check for obvious errors in parms */
+	if (key == NULL)
+		return EINVAL;
+
 #ifdef ICA_FIPS
 	if (fips >> 1)
 		return EACCES;
-	if (!curve_supported_via_openssl(key->nid))
+	if (!curve_supported_via_openssl(key->nid) || !curve_supported_via_cpacf(key->nid))
```

Comment:
I know, that `fips >> 1` has been checked in the line before, but maybe you can use `check_fips()` here and in all places below.